gradio/utils.py

```diff
@@ -465,6 +465,24 @@ def get_default_args(func: Callable) -> list[Any]:
     ]
 
 
+def safe_deepcopy(obj: Any) -> Any:
+    try:
+        return copy.deepcopy(obj)
+    except Exception:
+        if isinstance(obj, dict):
```

Comment:
We really only care about deep-copying the `props` dict right? I think we can skip the deep-copy for all other cases

Reply:
Ah yeah we can probably optimize, let me check

Reply:
I considered changing this, but I think its better to leave as is. There are other fields of the config that we want to copy, like `username` and `root`. Currently, they are primitives so we don't have to worry about deepcopying them, but in case they change in the future, then the safer approach is to deepcopy the whole config. Deepcopying the config only takes about 4 ms for a relatively complex demo like `kitchen_sink`, so performance should be ok